templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -687,6 +686,33 @@ CRM.$(function($) {
 {/literal}{/if}{literal}
 
 CRM.$(function($) {
+  onPaymentMethodChange();
+  $("#payment_instrument_id").on("change",function(){
+    onPaymentMethodChange();
+  });
+
+  function onPaymentMethodChange() {
+    var paymentInstrument = $('#payment_instrument_id').val();
```

Comment:
@eileenmcnaughton @pradpnayak wrt the point about making assumptions in the js about how various payment processors will do things, I would be happy if our new code hangs on the existing code displaying card type images. I don't feel we should be responsible for refactoring that, or not making use of it, @eileenmcnaughton